### ISSUE TITLE
Requirement of ImageNet folder structure and output validation

### ISSUE THREAD

**Issue by lcmeng** [yitu-opensource/T2T-ViT#24]
Hi,

Thank you for sharing your work. I tried to reproduce some of your results by
`CUDA_VISIBLE_DEVICES=0,1,2,3 ./distributed_train.sh 4 ~/workspace/datasets/ILSVRC2012 --model T2t_vit_7 -b 128 --lr 1e-3 --weight-decay .03 --cutmix 0.0 --reprob 0.25 --amp --img-size 224`

However, the `train_loss` and `eval_loss` in `summary.csv` both seem to be wrong. I saw in another issue that the ImageNet folders needs to have the right structure. In my case, I have `path_to_imagenet/train/train/*.jpeg` and `path_to_imagenet/val/{labelled_subgroups}/*.jpeg`. Do they agree with your setup? 

FYI, for the `train` folder, I also tried `path_to_imagenet/train/{labelled_subgroups}/*.jpeg`, which lead to seg fault...

```
epoch,train_loss,eval_loss,eval_top1,eval_top5
0,3.0369037985801697,6.954891597442627,0.094,0.504
epoch,train_loss,eval_loss,eval_top1,eval_top5
0,3.0369037985801697,6.954891597442627,0.094,0.504
epoch,train_loss,eval_loss,eval_top1,eval_top5
0,3.0369037985801697,6.954891597442627,0.094,0.504
epoch,train_loss,eval_loss,eval_top1,eval_top5
0,3.0364759885347805,6.954892452392578,0.094,0.504
epoch,train_loss,eval_loss,eval_top1,eval_top5
0,3.0364759885347805,6.954892452392578,0.094,0.504
epoch,train_loss,eval_loss,eval_top1,eval_top5
0,3.0364759885347805,6.954892452392578,0.094,0.504
epoch,train_loss,eval_loss,eval_top1,eval_top5
1,1.0352478187817793,6.95074588470459,0.094,0.524
epoch,train_loss,eval_loss,eval_top1,eval_top5
1,1.0352478187817793,6.95074588470459,0.094,0.524
epoch,train_loss,eval_loss,eval_top1,eval_top5
1,1.0352478187817793,6.95074588470459,0.094,0.524
epoch,train_loss,eval_loss,eval_top1,eval_top5
2,1.0149101408628316,6.952154161071777,0.102,0.54
epoch,train_loss,eval_loss,eval_top1,eval_top5
2,1.0149101408628316,6.952154161071777,0.102,0.54
epoch,train_loss,eval_loss,eval_top1,eval_top5
2,1.0149101408628316,6.952154161071777,0.102,0.54
epoch,train_loss,eval_loss,eval_top1,eval_top5
3,1.0148853338681734,6.9690971908569335,0.1,0.508
epoch,train_loss,eval_loss,eval_top1,eval_top5
3,1.0148853338681734,6.9690971908569335,0.1,0.508
epoch,train_loss,eval_loss,eval_top1,eval_top5
3,1.0148853338681734,6.9690971908569335,0.1,0.508
epoch,train_loss,eval_loss,eval_top1,eval_top5
4,1.0148665721599872,6.998007564697265,0.1,0.524
epoch,train_loss,eval_loss,eval_top1,eval_top5
4,1.0148665721599872,6.998007564697265,0.1,0.524
epoch,train_loss,eval_loss,eval_top1,eval_top5
4,1.0148665721599872,6.998007564697265,0.1,0.524
epoch,train_loss,eval_loss,eval_top1,eval_top5
5,1.0148564875125885,7.028750126037598,0.1,0.492
epoch,train_loss,eval_loss,eval_top1,eval_top5
5,1.0148564875125885,7.028750126037598,0.1,0.492
epoch,train_loss,eval_loss,eval_top1,eval_top5
5,1.0148564875125885,7.028750126037598,0.1,0.492
epoch,train_loss,eval_loss,eval_top1,eval_top5
6,1.0148570147844462,7.063653251342774,0.1,0.488
epoch,train_loss,eval_loss,eval_top1,eval_top5
6,1.0148570147844462,7.063653251342774,0.1,0.488
epoch,train_loss,eval_loss,eval_top1,eval_top5
6,1.0148570147844462,7.063653251342774,0.1,0.488
epoch,train_loss,eval_loss,eval_top1,eval_top5
7,1.0176889414970691,7.100188631896972,0.1,0.504
epoch,train_loss,eval_loss,eval_top1,eval_top5
7,1.0176889414970691,7.100188631896972,0.1,0.504
epoch,train_loss,eval_loss,eval_top1,eval_top5
7,1.0176889414970691,7.100188631896972,0.1,0.504
epoch,train_loss,eval_loss,eval_top1,eval_top5
8,1.0148765375980964,7.139705660552979,0.1,0.49
epoch,train_loss,eval_loss,eval_top1,eval_top5
8,1.0148765375980964,7.139705660552979,0.1,0.49
epoch,train_loss,eval_loss,eval_top1,eval_top5
8,1.0148765375980964,7.139705660552979,0.1,0.49
epoch,train_loss,eval_loss,eval_top1,eval_top5
9,1.014855329806988,7.1943902149963375,0.1,0.5
......
115,1.0148504261787121,9.212117127990723,0.1,0.5
epoch,train_loss,eval_loss,eval_top1,eval_top5
115,1.0148504261787121,9.212117127990723,0.1,0.5
epoch,train_loss,eval_loss,eval_top1,eval_top5
115,1.0148504261787121,9.212117127990723,0.1,0.5
epoch,train_loss,eval_loss,eval_top1,eval_top5
116,1.0148504032538488,9.212117127990723,0.1,0.5
epoch,train_loss,eval_loss,eval_top1,eval_top5
116,1.0148504032538488,9.212117127990723,0.1,0.5
epoch,train_loss,eval_loss,eval_top1,eval_top5
116,1.0148504032538488,9.212117127990723,0.1,0.5
```

Some detailed information is also logged:

```
Training in distributed mode with multiple processes, 1 GPU per process. Process 3, total 4.
adopt performer encoder for tokens-to-token
Training in distributed mode with multiple processes, 1 GPU per process. Process 1, total 4.
adopt performer encoder for tokens-to-token
Training in distributed mode with multiple processes, 1 GPU per process. Process 2, total 4.
adopt performer encoder for tokens-to-token
Training in distributed mode with multiple processes, 1 GPU per process. Process 0, total 4.
adopt performer encoder for tokens-to-token
Model T2t_vit_7 created, param count: 4310222
Data processing configuration for current model + dataset:
        input_size: (3, 224, 224)
        interpolation: bicubic
        mean: (0.485, 0.456, 0.406)
        std: (0.229, 0.224, 0.225)
        crop_pct: 0.9
Selected optimization level O1:  Insert automatic casts around Pytorch functions and Tensor methods.

Defaults for this optimization level are:
enabled                : True
opt_level              : O1
cast_model_type        : None
patch_torch_functions  : True
keep_batchnorm_fp32    : None
master_weights         : None
loss_scale             : dynamic
Processing user overrides (additional kwargs that are not None)...
After processing overrides, optimization options are:
enabled                : True
opt_level              : O1
cast_model_type        : None
patch_torch_functions  : True
keep_batchnorm_fp32    : None
master_weights         : None
loss_scale             : dynamic
Using NVIDIA APEX AMP. Training in mixed precision.
Using NVIDIA APEX DistributedDataParallel.
Scheduled epochs: 310
Gradient overflow.  Skipping step, loss scaler 0 reducing loss scale to 32768.0
Gradient overflow.  Skipping step, loss scaler 0 reducing loss scale to 32768.0
Gradient overflow.  Skipping step, loss scaler 0 reducing loss scale to 32768.0
Gradient overflow.  Skipping step, loss scaler 0 reducing loss scale to 32768.0
Train: 0 [   0/2502 (  0%)]  Loss:  6.951978 (6.9520)  Time: 4.628s,  110.63/s  (4.628s,  110.63/s)  LR: 1.000e-06  Data: 2.494 (2.494)
Train: 0 [  50/2502 (  2%)]  Loss:  6.508403 (6.7302)  Time: 0.303s, 1690.50/s  (0.368s, 1391.41/s)  LR: 1.000e-06  Data: 0.018 (0.070)
Train: 0 [ 100/2502 (  4%)]  Loss:  6.072941 (6.5111)  Time: 0.307s, 1667.06/s  (0.325s, 1577.30/s)  LR: 1.000e-06  Data: 0.016 (0.045)
Train: 0 [ 150/2502 (  6%)]  Loss:  5.658792 (6.2980)  Time: 0.288s, 1774.90/s  (0.311s, 1647.29/s)  LR: 1.000e-06  Data: 0.046 (0.036)
Train: 0 [ 200/2502 (  8%)]  Loss:  5.238940 (6.0862)  Time: 0.290s, 1766.25/s  (0.304s, 1682.54/s)  LR: 1.000e-06  Data: 0.015 (0.032)
Train: 0 [ 250/2502 ( 10%)]  Loss:  4.865733 (5.8828)  Time: 0.275s, 1860.66/s  (0.301s, 1702.35/s)  LR: 1.000e-06  Data: 0.015 (0.030)
Gradient overflow.  Skipping step, loss scaler 0 reducing loss scale to 16384.0
Gradient overflow.  Skipping step, loss scaler 0 reducing loss scale to 16384.0
Gradient overflow.  Skipping step, loss scaler 0 reducing loss scale to 16384.0
Gradient overflow.  Skipping step, loss scaler 0 reducing loss scale to 16384.0
Train: 0 [ 300/2502 ( 12%)]  Loss:  4.504626 (5.6859)  Time: 0.293s, 1745.01/s  (0.297s, 1721.03/s)  LR: 1.000e-06  Data: 0.017 (0.028)
Train: 0 [ 350/2502 ( 14%)]  Loss:  4.201088 (5.5003)  Time: 0.261s, 1959.78/s  (0.296s, 1732.37/s)  LR: 1.000e-06  Data: 0.018 (0.027)
Train: 0 [ 400/2502 ( 16%)]  Loss:  3.931564 (5.3260)  Time: 0.275s, 1861.54/s  (0.294s, 1740.32/s)  LR: 1.000e-06  Data: 0.017 (0.026)
Train: 0 [ 450/2502 ( 18%)]  Loss:  3.716414 (5.1650)  Time: 0.271s, 1886.86/s  (0.293s, 1745.85/s)  LR: 1.000e-06  Data: 0.019 (0.025)
Train: 0 [ 500/2502 ( 20%)]  Loss:  3.524733 (5.0159)  Time: 0.254s, 2017.25/s  (0.292s, 1751.30/s)  LR: 1.000e-06  Data: 0.016 (0.025)
Train: 0 [ 550/2502 ( 22%)]  Loss:  3.373711 (4.8791)  Time: 0.303s, 1692.35/s  (0.292s, 1756.09/s)  LR: 1.000e-06  Data: 0.019 (0.024)
Train: 0 [ 600/2502 ( 24%)]  Loss:  3.254548 (4.7541)  Time: 0.276s, 1854.60/s  (0.291s, 1760.98/s)  LR: 1.000e-06  Data: 0.018 (0.024)
Train: 0 [ 650/2502 ( 26%)]  Loss:  3.150394 (4.6396)  Time: 0.255s, 2004.50/s  (0.290s, 1763.84/s)  LR: 1.000e-06  Data: 0.018 (0.024)
Train: 0 [ 700/2502 ( 28%)]  Loss:  3.068058 (4.5348)  Time: 0.308s, 1664.82/s  (0.290s, 1767.39/s)  LR: 1.000e-06  Data: 0.017 (0.023)
......
Train: 117 [1850/2502 ( 74%)]  Loss:  1.014853 (1.0149)  Time: 0.311s, 1645.79/s  (0.285s, 1793.36/s)  LR: 6.727e-04  Data: 0.014 (0.022)
Train: 117 [1900/2502 ( 76%)]  Loss:  1.014856 (1.0149)  Time: 0.255s, 2004.47/s  (0.285s, 1794.04/s)  LR: 6.727e-04  Data: 0.015 (0.022)
Train: 117 [1950/2502 ( 78%)]  Loss:  1.014849 (1.0149)  Time: 0.307s, 1669.59/s  (0.285s, 1794.47/s)  LR: 6.727e-04  Data: 0.018 (0.022)
Train: 117 [2000/2502 ( 80%)]  Loss:  1.014849 (1.0149)  Time: 0.314s, 1631.92/s  (0.285s, 1795.23/s)  LR: 6.727e-04  Data: 0.017 (0.022)
Train: 117 [2050/2502 ( 82%)]  Loss:  1.014849 (1.0149)  Time: 0.276s, 1855.58/s  (0.285s, 1795.60/s)  LR: 6.727e-04  Data: 0.021 (0.022)
Train: 117 [2100/2502 ( 84%)]  Loss:  1.014851 (1.0149)  Time: 0.271s, 1888.39/s  (0.285s, 1796.09/s)  LR: 6.727e-04  Data: 0.016 (0.022)
Train: 117 [2150/2502 ( 86%)]  Loss:  1.014849 (1.0149)  Time: 0.253s, 2022.58/s  (0.285s, 1796.20/s)  LR: 6.727e-04  Data: 0.017 (0.022)
Train: 117 [2200/2502 ( 88%)]  Loss:  1.014849 (1.0149)  Time: 0.284s, 1802.80/s  (0.285s, 1796.40/s)  LR: 6.727e-04  Data: 0.016 (0.021)
Train: 117 [2250/2502 ( 90%)]  Loss:  1.014850 (1.0149)  Time: 0.470s, 1089.05/s  (0.285s, 1796.49/s)  LR: 6.727e-04  Data: 0.014 (0.021)
Train: 117 [2300/2502 ( 92%)]  Loss:  1.014848 (1.0149)  Time: 0.255s, 2007.12/s  (0.285s, 1797.02/s)  LR: 6.727e-04  Data: 0.017 (0.021)
Train: 117 [2350/2502 ( 94%)]  Loss:  1.014852 (1.0149)  Time: 0.274s, 1870.79/s  (0.285s, 1797.39/s)  LR: 6.727e-04  Data: 0.016 (0.021)
Train: 117 [2400/2502 ( 96%)]  Loss:  1.014855 (1.0149)  Time: 0.314s, 1629.84/s  (0.285s, 1797.29/s)  LR: 6.727e-04  Data: 0.017 (0.021)
Train: 117 [2450/2502 ( 98%)]  Loss:  1.014856 (1.0149)  Time: 0.289s, 1769.74/s  (0.285s, 1797.42/s)  LR: 6.727e-04  Data: 0.017 (0.021)
```

**Comment by yuanli2333** [comment]
Hi, the correct data strcuture should be:

```
imagenet/
....train/
........calss1/
............/ img1
........class2/
............/ img2
....val/
........calss1/
............/ img1
........class2/
............/ img2
```
You should preprocess you data strcuture as the above, then you can get correct training and validation loss.
I will update this to our repo.

**Comment by lcmeng** [comment]
Thank you for the quick reply. I mentioned in my earlier post that "for the train folder, I also tried `path_to_imagenet/train/{labelled_subgroups}/*.jpeg`, which lead to seg fault...". This layout is the same as what you described, so I checked again and found a sub-directory that breaks the convention due to a previous project. 

So all is good. The GPUs are burning as expected now. Thank you. I'll close this issue.
